### PR TITLE
Themed development environments

### PR DESCRIPTION
**Pull request by BradyMitch**
# Description
[//]: # (THESE COMMENTS ARE HIDDEN WHEN THE PR IS PUBLISHED, no need to remove them.)
[//]: # (Please include a summary of the changes.)

Makes a visual change to each environment (local, dev, test, prod) as seen in the images below.

local

dev

test

[](https://hamzamohdzubair.github.io/redant/)

[//]: # (THESE COMMENTS ARE HIDDEN WHEN THE PR IS PUBLISHED, no need to remove them.)
[//]: # (Please include a list of Jira tickets and or Issues this PR resolves.)

- [PIMS-88](https://apps.itsm.gov.bc.ca/jira/browse/PIMS-88)

[](https://hamzamohdzubair.github.io/redant/)

[//]: # (THESE COMMENTS ARE HIDDEN WHEN THE PR IS PUBLISHED, no need to remove them.)
[//]: # (Please include a list of known issues if there are any.)

- None listed.

## Type of change
[//]: # (THESE COMMENTS ARE HIDDEN WHEN THE PR IS PUBLISHED, no need to remove them.)
[//]: # (Please delete options that are not relevant.)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Dependency updates (may include refactoring of a dependencies functionality)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
[//]: # (THESE COMMENTS ARE HIDDEN WHEN THE PR IS PUBLISHED, no need to remove them.)
[//]: # (Please describe the tests that you ran to verify your changes.)

Manually changed the variable that switches the themes.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Sharala-Perumal** [comment]
Hi Brady, can you try with little lighter colours but pleasant at the same time? The dark colors are still hard to differentiate b/w environments.

**Comment by BradyMitch** [comment]
> Hi Brady, can you try with little lighter colours but pleasant at the same time? The dark colors are still hard to differentiate b/w environments.

I had tried with lighter colours but it didnt really fit.

File: frontend/src/components/layout/Header/Header.tsx [hunk 1]
```diff
@@ -27,6 +27,12 @@ const Header = () => {
   const network = useNetworkStore();
   const { requests } = useAppSelector(store => store.network);
 
+  // Change styling based on environment
+  let environment = 'production';
+  if (window.location.href.includes('localhost')) environment = 'local';
+  else if (window.location.href.includes('dev')) environment = 'development';
+  else if (window.location.href.includes('test')) environment = 'testing';
```
Comment (ZachBourqueBCGov) [inline frontend/src/components/layout/Header/Header.tsx:34]: Can we change this logic to use the environment variable NODE_ENV?

Comment (ZachBourqueBCGov) [reply frontend/src/components/layout/Header/Header.tsx:34]: In which case we could omit the "environment" variable completely and use process.env.NODE_ENV

Comment (Fosol) [reply frontend/src/components/layout/Header/Header.tsx:34]: I believe you have to change the build/deploy in Openshift to do this.

Comment (BradyMitch) [reply frontend/src/components/layout/Header/Header.tsx:34]: Ya the pipeline isn't really suited for that change at the moment. We would have to change any case in the code that already checks NODE_ENV as its expecting either 'production' or 'development' and wouldn't expect 'local' or 'testing'. Also if we were to change these environment variables within BuildConfigs we'd have to modify the deploy pipeline as it doesnt create a new image, it just deploys the image from the test environment.

Comment (ZachBourqueBCGov) [reply frontend/src/components/layout/Header/Header.tsx:34]: I'll put a ticket in the backlog to look into this. Each environment should explicitly know what environment it is

**Review by ZachBourqueBCGov — APPROVED** [review]

**Pull request merged**